File: src/main/prefs.ts
/**
 * Small JSON store for non-secret user choices (selected business
 * policies). Kept separate from config: these are picked in the UI at
 * runtime, not configured in .env.
 */

import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'node:fs'
import { dirname, join } from 'node:path'

import { app } from 'electron'

import type { PolicyIds } from '@shared/types'

interface Prefs {
  policyIds?: PolicyIds
}

function prefsPath(): string {
  return join(app.getPath('userData'), 'prefs.json')
}

function readPrefs(): Prefs {
  const path = prefsPath()
  if (!existsSync(path)) return {}
  try {
    return JSON.parse(readFileSync(path, 'utf-8')) as Prefs
  } catch {
    return {}
  }
}

function writePrefs(prefs: Prefs): void {
  const path = prefsPath()
  mkdirSync(dirname(path), { recursive: true })
  writeFileSync(path, JSON.stringify(prefs, null, 2), 'utf-8')
}

export function getPolicyIds(): PolicyIds {
  return readPrefs().policyIds ?? {}
}

export function setPolicyIds(policyIds: PolicyIds): void {
  writePrefs({ ...readPrefs(), policyIds })
}

export function hasCompletePolicies(ids: PolicyIds): boolean {
  return Boolean(ids.fulfillment && ids.payment && ids.return)
}

File: src/epay_tool/prefs.py
"""Small JSON store for non-secret user choices (selected business policies).

Kept separate from Settings: these are picked in the UI at runtime, not
configured in .env.
"""

from __future__ import annotations

import json
from pathlib import Path

from epay_tool.config import DATA_DIR, ensure_dirs

PREFS_PATH: Path = DATA_DIR / "prefs.json"


def load_prefs() -> dict:
    if not PREFS_PATH.exists():
        return {}
    try:
        return json.loads(PREFS_PATH.read_text(encoding="utf-8"))
    except (json.JSONDecodeError, OSError):
        return {}


def save_prefs(prefs: dict) -> None:
    ensure_dirs()
    PREFS_PATH.write_text(json.dumps(prefs, indent=2, ensure_ascii=False), encoding="utf-8")


def get_policy_ids() -> dict[str, str]:
    return load_prefs().get("policy_ids", {})


def set_policy_ids(policy_ids: dict[str, str]) -> None:
    prefs = load_prefs()
    prefs["policy_ids"] = policy_ids
    save_prefs(prefs)
